src/components/Common/UI/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,101 @@
+import { useId } from 'react'
+import type { FocusEventHandler, InputHTMLAttributes, Ref } from 'react'
+import {
+  Input,
+  Group,
+  // eslint-disable-next-line no-restricted-imports
+  NumberField as AriaNumberField,
+} from 'react-aria-components'
+import { FieldLayout, type FieldLayoutProps } from '../FieldLayout'
+import { useLocale } from '@/contexts/LocaleProvider'
+
+export interface NumberInputProps extends FieldLayoutProps {
+  name?: string
+  format?: 'currency' | 'decimal' | 'percent'
+  currencyDisplay?: keyof Intl.NumberFormatOptionsCurrencyDisplayRegistry | undefined
+  minValue?: number
```

Comment:
We don't have to do this right now, but we've chatted before about picking the props from the React HTML Type for Number Inputs as an example, should we be thinking about that?

Reply:
Yeah good point, maybe the best way forward is to pick from the html prop types. let me do some experimentation with it